### ISSUE TITLE
Is reward_fn equal to log_softmax

### ISSUE THREAD

**Issue by EganGu** [microsoft/LMOps#222]
I noticed that the `scores` in `reward_fn` is actually equal to `logits_i - logsumexp(logits)`.
I think this expression can be calculated directly by `log_softmax`. Why not use `log_softmax`? 

https://github.com/microsoft/LMOps/blob/5fbf5bcd6e6760fa95aaaf945fb5d9cb033135f6/minillm/minillm/reward.py#L33

**Comment by t1101675** [comment]
This is because for models with tensor parallelism, `log_softmax` should be computed as `logits_i - logsumexp(logits)`. To maintain consistency and compare the results, we also use `logits_i - logsumexp(logits)` in normal scenarios.

**Comment by EganGu** [comment]
> This is because for models with tensor parallelism, `log_softmax` should be computed as `logits_i - logsumexp(logits)`. To maintain consistency and compare the results, we also use `logits_i - logsumexp(logits)` in normal scenarios.

Understood. Thanks for your reply.
